presto-mongodb/src/main/java/com/facebook/presto/mongodb/MongoPageSource.java

```diff
@@ -177,11 +178,22 @@ else if (javaType == Block.class) {
         }
     }
 
+    private String toVarcharValue(Object value)
+    {
+        if (value instanceof Collection<?>) {
+            return "[" + String.join(", ", ((Collection<?>) value).stream().map(this::toVarcharValue).collect(toList())) + "]";
+        }
+        if (value instanceof Document) {
+            return ((Document) value).toJson();
+        }
+        return value.toString();
```

Comment:
would that ever be the case?

Reply:
It is possible that the value is not a Document. For example, if the value is just a string or a primitive. We should still support converting that to varchar.

Reply:
That said, I realized I may need to change value.toString() to String.valueOf(value) to avoid NPE.

Reply:
Could you add a test for primitive type?

Reply:
Added both tests (primitives, array of primitives with null value).